docs/SystemDescriptionDocument.md

```diff
@@ -0,0 +1,74 @@
+# Overview
+The goal of this project is to verify off-chain EVM computation on-chain.
+In our particular case, it is used to correctly exit spending conditions from a side-chain to the main-chain
+and also a means of a PoS mechanism to secure the side-chain.
+
+Executing a program consists of many intermediate steps and in turn,
+the EVM consists of various elements which change their data after each single step (opcode) of the program.
+These intermediate states can be compared in an deterministic manner, by hashing the state of the EVM’s elements into a single state hash.
+This makes it possible to construct a merkle tree from all those state hashes, each one of them representing a single execution step.
+
+To find the first convergence of the execution (the first mismatch), we start a round-based path walking,
+starting from the merkle root until we reach the leaves.
+
+For more information on how this works, take a look at [Merkelizer.md](./Merkelizer.md).
+
+## Example
+Let us assume a solver registers a execution result given the `code` to the program, any inputs (`callData`) and the merkle root hash of the execution.
```

Comment:
a execution -> an execution
the code to the program -> the code of the program
I think 'any' can by dropped?